core/src/Cache/CacheFallbackOperations.cs

```diff
@@ -96,9 +107,21 @@ internal class CacheFallbackOperations
                     return;
                 }
 
+                if (!String.Equals(rtItem?.Environment, idItem?.Environment, StringComparison.OrdinalIgnoreCase))
+                {
+                    CoreLoggerBase.Default.Error(DifferentEnvError);
+                    CoreLoggerBase.Default.ErrorPii(DifferentEnvError);
+                }
+
+                if (!String.Equals(rtItem?.Environment, (new Uri(authority)).Host, StringComparison.OrdinalIgnoreCase))
+                {
+                    CoreLoggerBase.Default.Error(DifferentAuthorityError);
+                    CoreLoggerBase.Default.ErrorPii(DifferentAuthorityError);
+                }
+
```

Comment:
Have we seen any issues like this ever occur? Is it something critical? Can we consider only doing this validation in case there is not token returned or an error happens?
Goal: Only do these string operations if really needed (if things are found we should be good)
e.g. is it possible to only do this work as part of the exception handler?